dec2024/client.js

```diff
@@ -0,0 +1,92 @@
+require("isomorphic-fetch");
+
+class Client {
+  constructor({ hasuraSecret, targetURL }) {
+    this.secret = hasuraSecret;
+    this.url = targetURL;
+  }
+
+  async graphQL(query, variables) {
+    const result = await fetch(this.url, {
+      method: "POST",
+      body: JSON.stringify({ query, variables }),
+      headers: {
+        "x-hasura-admin-secret": this.secret,
+      },
+    });
+    return result.json();
+  }
+
+  async getQueuedFlow() {
+    const { data, errors } = await this.graphQL(
+      `query GetQueuedFlow {
+        temp_data_migrations_audit(limit: 1, where: {updated: {_eq: false}}) {
+          id: flow_id
+          flow: flows {
+            slug
+            team {
+              slug
+            }
+            data
+            publishedFlows: published_flows(limit: 1, order_by: {created_at: desc}) {
+              id  
+              data
+            }
+          }
+        }
+      }`
+    );
+
+    if (errors || !data) throw new Error(formatJSON({ data, errors }));
+    return data.temp_data_migrations_audit[0];
+  }
+
+  async updateQueuedFlow(flowId, liveFlowData) {
+    const { data, errors } = await this.graphQL(
+      `mutation UpdateFlow ($flowId: uuid!, $liveData: jsonb!) {
+        update_flows_by_pk(pk_columns: {id: $flowId}, _set: {data: $liveData}) {
+          id
+        }
+        update_temp_data_migrations_audit_by_pk(pk_columns: {flow_id: $flowId}, _set: {updated: true}) {
+          flow_id
```

Comment:
Totally not essential, but it could be nice to capture the processed log here also. Rather than just console.log() as we process a flow, we could save to an object/array and keep as a record in the audit table. Something for next time round maybe!

Reply:
Yep this is a good thought & would be more consistent - 
- When running locally, you'll just get console logs
- When running via crontab, I'm writing all logs to a single file (good record-keeping and easy to ctrl+f search, but disjointed from audit table!)